### PR TITLE
Gumdrop/cmv2

### PR DESCRIPTION
**Pull request by lwus**
- update gumdrop candy-machine integration to do a whitelist-token drop
- cmv2 claim workflow is
    - claim whitelist token(s) gumdrop
    - crank cmv2 mint-nft
- revamp example frontend
    - rip out web-based gumdrop creation and closing
    - make UI more similar to metaplex storefront and some clean up of claim input fields

### CODE REVIEW

**Comment by stegaBOB** [comment]
Can you run `yarn lint:fix` on this

**Comment by lwus** [comment]
I already ran that. Seems to be another issue https://github.com/metaplex-foundation/metaplex/runs/5195094909?check_suite_focus=true#step:5:10

**Comment by stegaBOB** [comment]
#1795 should fix this (assuming my YAML is accurate)

File: js/packages/cli/src/gumdrop-cli.ts [hunk 1]
```diff
@@ -470,6 +526,42 @@ programCommand('get_contact')
...
+      options.rpcUrl || anchor.web3.clusterApiUrl(options.env),
+    );
+
+    const programPointer = await connection.getAccountInfo(
+      GUMDROP_DISTRIBUTOR_ID,
+    );
+    const executableBufferKey = new PublicKey(programPointer.data.slice(4));
+
+    const programBuffer = await connection.getAccountInfo(executableBufferKey);
+    const programData = programBuffer.data.slice(0x30 - 3); // by inspection lol...
```
Comment (stegaBOB) [inline js/packages/cli/src/gumdrop-cli.ts:548]: lol

**Comment by stegaBOB** [comment]
Going to run some more tests and set up a gumdrop tomorrow. Hopefully get this jumbo PR merged in over the next couple of days.

File: js/packages/cli/src/gumdrop-cli.ts (hunk 1 above)
Comment (lwus) [reply js/packages/cli/src/gumdrop-cli.ts:548]: OK size of https://github.com/solana-labs/solana/blob/master/sdk/program/src/bpf_loader_upgradeable.rs#L41 (4 + 8 + 1 + 32)

Comment (stegaBOB) [reply js/packages/cli/src/gumdrop-cli.ts:548]: I'm just amused at the comment haha. "by inspection lol"

**Comment by stegaBOB** [comment]
Trying to review this and I'm struggling a bit to get everything created. The docs seem quite a bit outdated too. I think it's going to be really challenging for non technical users to create a gumdrop without the create UI. For now it's better than being completely broken though.

**Comment by stegaBOB** [comment]
@lwus is there a reason you removed the web create gumdrop section?

**Comment by lwus** [comment]
- The create web UI was becoming more and more dated and missed a few features anyway (e.g Discord). Do you know what the ratio of people using the web UI vs CLI is?
- The docs have not been updated with these changes but I can tackle that at some point
- Integration testing AWS / whatnot is probably a bit tricky (not sure how frequently these features are used anyway) but I'll add spinning up some simple testing setups to my todo list. Is that a blocker for this?

Apologies for the delay!

**Comment by stegaBOB** [comment]
@lwus @austbot The big thing is the lack of documentation at the moment I think. Its quite a bit daunting to do everything via the CLI right now with so many options and not much info on what is needed and how to do it.

**Comment by stegaBOB** [comment]
I personally couldn't figure it out without actually looking into the code.

**Comment by stegaBOB** [comment]
@lwus would you be able to update the docs for all these changes in our docs.metaplex.com? I opened an issue there https://github.com/metaplex-foundation/docs/issues/187

**Comment by stegaBOB** [comment]
Checking the docs! On a plane to Austin right now but I'll fix the conflicts/do some more testing and hopefully get this merged in today or tomorrow.

**Comment by Sense** [comment]
> Checking the docs! On a plane to Austin right now but I'll fix the conflicts/do some more testing and hopefully get this merged in today or tomorrow.

Any news on this PR?
much appreciated, thanks for your work 🔥

**Comment by stegaBOB** [comment]
The new repo is https://github.com/metaplex-foundation/gumdrop. Thanks again @lwus! :heart:

**Pull request closed without merging**